pyro/contrib/autoguide/__init__.py

```diff
@@ -561,6 +564,51 @@ def _loc_scale(self, *args, **kwargs):
         return loc, scale
 
 
+class AutoTransformedNormal(AutoContinuous):
+    """
+    This implementation of :class:`AutoContinuous` uses a Transformed
+    Low Rank MV Normal distribution via a IAF to construct a guide
+    over the entire latent space. The guide does not depend on the model's
+    ``*args, **kwargs``.
+
+    Usage::
+
+        guide = AutoTransformedNormal(model, rank=10)
+        svi = SVI(model, guide, ...)
+
+    :param callable model: a generative model
+    :param int rank: the rank of the low-rank part of the covariance matrix
+    :param str prefix: a prefix that will be prefixed to all param internal sites
+    """
+    def __init__(self, model, hidden_dim=None, sigmoid_bias=2.0, prefix="auto", rank=1):
+        if not isinstance(rank, numbers.Number) or not rank > 0:
+            raise ValueError("Expected rank >= 0 but got {}".format(rank))
+        self.sigmoid_bias = sigmoid_bias
+        self.hidden_dim = hidden_dim
+        self.rank = rank
+        super(AutoTransformedNormal, self).__init__(model, prefix)
+
+    def sample_latent(self, *args, **kwargs):
+        """
+        Samples the (single) multivariate normal latent used in the auto guide.
+        """
+        hidden_dim = self.hidden_dim if self.hidden_dim is not None else self.latent_dim
```

Comment:
In might be simpler to
```py
if self.hidden_dim is None:
    self.hidden_dim = self.latent_dim
```
and thereafter refer only to `self.hidden_dim`.

Reply:
yeah i did that at first then didnt want masked errors if `_loc_scale` is called before `sample_latent` since `self.latent_dim` is only constructed during `__call__`

Reply:
I think we should just error if `_loc_scale()` is called before `__call__()`